File: Data Structures & Algorithms/two-integer-sum/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let m = new Map();

        for (let i=0; i<nums.length; i++) {
            m.set(nums[i], i);
        }

        let res = [];
        for(let i=0; i<nums.length; i++){
            let r = target - nums[i];
            if (m.has(r) && m.get(r) !== i){
                res.push(m.get(r));
                res.push(i);
                break;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-6.cpp
class Solution {
public:
    bool isValid(string s) {
        string res{};
        for(int i=0; i<s.size(); i++){
            if (s[i] == '(' || s[i] == '[' || s[i] == '{'){
                res += s[i];
                cout << res << '\n';
            } else {
                if (res.size() == 0){
                    return false;
                } else if (s[i] == ')' && res[res.size()-1] != '('){
                    return false;
                } else if (s[i] == '}' && res[res.size()-1] != '{'){
                    return false;
                } else if (s[i] == ']' && res[res.size()-1] != '['){
                    return false;
                } else {
                    res.pop_back();
                }
            }
        }
        cout << res;

        return res.size() == 0;
    }
};
